flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/generated/CompilationOption.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.generated;
+
+import scala.Dynamic;
+
+/**
+ * The option for compiling generated Java code.
+ */
+public enum CompilationOption {
+
+	/**
+	 * Compiling Java code by Janino.
+	 * The compilation is fast, but the generated binary code is of low quality.
+	 */
+	JANINO,
+
+	/**
+	 * Compiling Java code by Java Compiler API (JCA)
+	 * The compilation is slow, but the generated binary code is of high quality.
```

Comment:
Are there any proofs that JCA is always better than janino, I think for long running jobs, JIT is the key for performance instead of fresh generated code. And more tests may need for this conclusion.

Reply:
@Xpray , thanks a lot for your comments.
We have some proofs that JCA can be better than Janino. For example, when trying to vectorize Flink, we found that by using JCA, the performance of TPC-H Q1 (1 TB) can be improved by over 20% (27-28s to 20s). Similar performance improvements are also witness for other queries like Q12 and Q18.

However, we cannot claim that JCA always outperforms Janino. Because JCA takes longer time for compiling, the performance of JCA can even be worse in some cases. 

Therefore, the meaning of this issue is not to give users a guarantee of better performance. Instead, we give the users a new choice, which may help them in their specific scenarios.

Reply:
If that is the case, I think these comments ' The compilation is slow, but the generated binary code is of high quality' should be removed.

Reply:
Removed. Thanks for your suggestion.